### PR TITLE
adds two new infoblox lookup plugins

### PR DESCRIPTION
**Pull request by privateip**
* nios - lookup plugin to return nios objects to the playbook
* nios_next_ip - lookup plugin to return the next avaiable ip address

### CODE REVIEW

File: lib/ansible/plugins/lookup/nios.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
+#
```
Comment (gundalow) [inline lib/ansible/plugins/lookup/nios.py:1]: Please add to `.github/BOTMETA.yml` like https://github.com/ansible/ansible/pull/35179/files
Will ensure we get notified about bugs and PRs

File: lib/ansible/utils/module_docs_fragments/nios.py [hunk 2]
```diff
@@ -31,35 +31,49 @@ class ModuleDocFragment(object):
         description:
           - Specifies the DNS host name or address for connecting to the remote
             instance of NIOS WAPI over REST
+          - Value can also be specified using INFOBLOX_HOST environment
```
Comment (gundalow) [inline lib/ansible/utils/module_docs_fragments/nios.py]: `C(INFOBLOX_HOST)`

Same for other ENVs

File: lib/ansible/plugins/lookup/nios_next_ip.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+from ansible.errors import AnsibleError
+
+
+class LookupModule(LookupBase):
+
+    def run(self, terms, variables=None, **kwargs):
+        try:
+            network = terms[0]
+        except IndexError:
+            raise AnsibleError('the network must be specified')
```
Comment (gundalow) [inline lib/ansible/plugins/lookup/nios_next_ip.py]: Can we say which network?

File: lib/ansible/plugins/lookup/nios_next_ip.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+---
+lookup: nios_next_ip
+version_added: "2.5"
+short_description: Return the next available IP address for a network
+description:
+  - Uses the Infoblox WAPI API to return the next available IP addresses
+    for a given network CIDR
+requirements:
+  - infoblox_client
+extends_documentation_fragment: nios
```
Comment (gundalow) [inline lib/ansible/plugins/lookup/nios_next_ip.py:33]: Not sure if `extends_documentation_fragments:` works. for `lookups`. I don't see it used in any other `lookups`

`make webdocs`
walk to store
buy coffee
Review docs

would confirm

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/lookup/nios_next_ip.py (hunk 4 above)
Comment (privateip) [reply lib/ansible/plugins/lookup/nios_next_ip.py:33]: confirmed works fine

**Review by gundalow — APPROVED** [review]

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Merged, CI failures not related

**Pull request merged**